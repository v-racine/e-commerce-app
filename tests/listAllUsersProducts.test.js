const { Config } = require('../src/config/config');

Config.Get(process.env);

const AppFactory = require('../src/app');
const request = require('supertest');

describe('list all products', () => {
  const mockProductsRepo = {};

  const app = AppFactory({
    productsRepo: mockProductsRepo,
  });

  describe('when: users want to view all products on the homepage', () => {
    let rsp;

    let products = [
      {
        id: '1234',
        title: 'product1',
        price: '12.00',
      },
      {
        id: '12345',
        title: 'product2',
        price: '101000.00',
      },
    ];

    beforeEach(async () => {
      mockProductsRepo.getAll = jest.fn().mockImplementation(() => {
        return new Promise((resolve) => {
          resolve(products);
        });
      });

      rsp = await request(app).get('/').send();
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    test('then: we return a list of all the products', async () => {
      const text = rsp.text;
      expect(text).toMatchSnapshot();

      expect(mockProductsRepo.getAll).toHaveBeenCalledTimes(1);

      const status = rsp.status;
      expect(status).toBe(200);
    });
  });
});
