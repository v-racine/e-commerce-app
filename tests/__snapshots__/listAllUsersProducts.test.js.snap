// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list all products when: users want to view all products on the homepage then: we return a list of all the products 1`] = `
"
    <!DOCTYPE html>
      <html lang="en">
      <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>Shop</title>
        
        <link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css" rel="stylesheet">
        <link href="/css/main.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bulma/0.7.5/css/bulma.min.css"></link>
      </head>

      <body>
        <header>
          <nav class="navbar navbar-top">
            <div class="container navbar-container">
              <div>
                <ul class="social">
                  <li>
                    <a href=""><i class="fa fa-phone"></i>+1 555 987 6543</a>
                  </li>
                  <li>
                    <a href=""><i class="fa fa-envelope"></i> shop@myshop.com</a>
                  </li>
                </ul>
              </div>
              <div>
                <ul class="social">
                  <li><a href=""><i class="fab fa-facebook"></i></a></li>
                  <li><a href=""><i class="fab fa-twitter"></i></a></li>
                  <li><a href=""><i class="fab fa-linkedin"></i></a></li>
                  <li><a href=""><i class="fab fa-dribbble"></i></a></li>
                  <li><a href=""><i class="fab fa-google-plus"></i></a></li>
                </ul>
              </div>
            </div>
          </nav>
          <nav class="navbar navbar-bottom">
            <div class="container navbar-container">
              <div>
                <a href="/">
                  <h3 class="title">EComm Shop</h3>
                </a>
              </div>
              <div class="navbar-item">
                <div class="navbar-buttons">
                  <div class="navbar-item">
                    <a href="/"><i class="fa fa-star"></i> Products</a>
                  </div>
                  <div class="navbar-item">
                    <a href="/cart"><i class="fa fa-shopping-cart"></i> Cart</a>
                  </div>
                </div>
              </div>
            </div>
          </nav>
        </header>

        
      <section class="banner">
        <div class="container">
          <div class="columns is-centered">
            <img src="/images/banner.jpg" />
          </div>
        </div>
      </section>
      
      <section>
        <div class="container">
          <div class="columns">
            <div class="column "></div>
            <div class="column is-four-fifths">
              <div>
                <h2 class="title text-center">Featured Items</h2>
                <div class="columns products">
                  
        <div class="column is-one-quarter">
          <div class="card product-card">
            <figure>
              <img src="data:image/png;base64, undefined"/>
            </figure>
            <div class="card-content">
              <h3 class="subtitle">product1</h3>
              <h5>$12.00</h5>
            </div>
            <footer class="card-footer">
              <form action="/cart/products" method="POST">
                <button class="button has-icon is-inverted">
                  <i class="fa fa-shopping-cart"></i> Add to cart
                </button>
              </form>
            </footer>
          </div>
        </div>
      

        <div class="column is-one-quarter">
          <div class="card product-card">
            <figure>
              <img src="data:image/png;base64, undefined"/>
            </figure>
            <div class="card-content">
              <h3 class="subtitle">product2</h3>
              <h5>$101000.00</h5>
            </div>
            <footer class="card-footer">
              <form action="/cart/products" method="POST">
                <button class="button has-icon is-inverted">
                  <i class="fa fa-shopping-cart"></i> Add to cart
                </button>
              </form>
            </footer>
          </div>
        </div>
        
                </div>
              </div>
            </div>
            <div class="column "></div>
          </div>
        </div>
      </section>
    
      </body>
    </html>
  "
`;
